modules/sandboxed-containers-deleting-resources.adoc

```diff
@@ -0,0 +1,25 @@
+//Module included in the following assemblies:
+//
+// *uninstalling-sandboxed-containers.adoc
+
+[id="sandboxed-containers-deleting-resources_{context}"]
+= Deleting {sandboxed-containers-first} resources
+
+You can remove and uninstall the `kata` runtime and all its related resources, such as CRI-O config and `RuntimeClass`, from your cluster.
+
+.Prerequisites
+
+* You have {product-title} {product-version} installed on your cluster.
+* You have installed the OpenShift CLI (`oc`).
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Delete the `KataConfig` custom resource by running the following command:
++
+[source,terminal]
+----
+oc delete kataconfig <KataConfig_CR_Name>
```

Comment:
To also get rid of the CRD we need to delete it with `oc delete crd kataconfigs.kataconfiguration.openshift.io`

Reply:
@jensfr what is CRD?

Reply:
The Custom Resource Definition. You can think of the CRD as a Class and CR (for example example-kataconfig) as an instance of this class. So we need to delete both the example-kataconfig CR and the class. 

When you only delete the example-kataconfig CR and then do 'oc get kataconfig' the reply will be 'no kataconfig instances exist' or something like that. After you deleted the CRD as well it will reply with 'unknown CR kataconfig'.

Reply:
@sagidlow I think that would be translated to a two-step process: 

1. Delete the `KataConfig` Custom Resouce (CR) (what we already have)
2. Delete the `KataConfig` Custom Resource Definition (CRD)

Can we add step 2?

Reply:
@jensfr Is it planned that the operator would do these things in the future? If so we need to remember that we will need to remove the steps from the doc.

(I'm a bit unsure if, in the general OpenShift philosophy, the shutdown of pods, removal of CR and CRD are supposed to be manual procedures or if the uninstallation procedure in the operator should do that for you)

Reply:
@c3d it's not automatable (if that's even a word :D) 

> When uninstalling an operator managed by OLM, a Cluster Admin must decide whether or not to remove the CustomResourceDefinitions (CRD), APIServices, and resources related to these types owned by the operator. By design, when OLM uninstalls an operator it does not remove any of the operator’s owned CRDs, APIServices, or CRs in order to prevent data loss. Instead, it is left to the Cluster Admin to remove any unwanted types and resources from the cluster. This document will discuss the steps a Cluster Admin should take when uninstalling an operator.

https://olm.operatorframework.io/docs/tasks/uninstall-operator/

Reply:
The OpenShift docs specifically say that pods using features from an operator should continue to work even after the operator is uninstalled (subscription deleted).  From https://docs.openshift.com/container-platform/4.7/operators/admin/olm-deleting-operators-from-cluster.html#olm-deleting-operators-from-a-cluster-using-web-console_olm-deleting-operators-from-a-cluster

"The Operator, any Operator deployments, and pods are removed by this action. Any resources managed by the Operator, including CRDs and CRs, are not removed. The web console enables dashboards and navigation items for some Operators. To remove these after uninstalling the Operator, you might need to manually delete the Operator CRDs."

Reply:
Ack to all the above. Thanks for verifying.